src/config.rs

```diff
@@ -1,30 +1,105 @@
-use std::collections::HashMap as Map;
+use std::collections::HashMap;
 use std::marker::PhantomData;
 use std::ops::Deref;
+use std::rc::Rc;
 use std::str::FromStr;
 
 use serde::de::{Deserialize, Deserializer};
 use serde_derive::Deserialize;
 use toml::value;
 
 use crate::de::*;
+use crate::errors;
 use crate::icons;
 use crate::input::MouseButton;
 use crate::themes::Theme;
 
+////////////////
+
+#[derive(Debug)]
+pub struct SharedConfig {
+    pub theme: Rc<Theme>,
+    pub icons: Rc<HashMap<String, String>>,
+    pub scrolling: Scrolling,
+}
+
+impl SharedConfig {
+    pub fn new(config: &Config) -> Self {
+        Self {
+            theme: Rc::new(config.theme.clone()),
+            icons: Rc::new(config.icons.clone()),
+            scrolling: config.scrolling,
+        }
+    }
+
+    pub fn theme_override(&mut self, overrides: &HashMap<String, String>) -> errors::Result<()> {
+        let mut theme = self.theme.as_ref().clone();
+        for entry in overrides {
+            match entry.0.as_str() {
+                "idle_fg" => theme.idle_fg = Some(entry.1.to_string()),
+                "idle_bg" => theme.idle_bg = Some(entry.1.to_string()),
+                "info_fg" => theme.info_fg = Some(entry.1.to_string()),
+                "info_bg" => theme.info_bg = Some(entry.1.to_string()),
+                "good_fg" => theme.good_fg = Some(entry.1.to_string()),
+                "good_bg" => theme.good_bg = Some(entry.1.to_string()),
+                "warning_fg" => theme.warning_fg = Some(entry.1.to_string()),
+                "warning_bg" => theme.warning_bg = Some(entry.1.to_string()),
+                "critical_fg" => theme.critical_fg = Some(entry.1.to_string()),
+                "critical_bg" => theme.critical_bg = Some(entry.1.to_string()),
+                x => {
+                    return Err(errors::ConfigurationError(
+                        format!("Theme element \"{}\" cannot be overriden", x),
```

Comment:
and allow everything to be overridden

Reply:
OK, but that also needs some changes in `print_blocks()` to actually use those overrides.

Reply:
And right now there is no way to get `shared_config` outside of the block itself. I have to add a new function to block trait `get_shared_config() -> SharedConfig`.

Reply:
Do we really need to have overriding of separator for each block? Anyway, I think I'll leave it for a separate PR since it's a lot of work.

Reply:
That's fine